core/services/ocrcommon/telemetry_test.go

```diff
@@ -204,5 +243,122 @@ func TestSendEATelemetry(t *testing.T) {
+	assert.Equal(t, obs, int64(0))
+	assert.Equal(t, logs.Len(), 1)
+	assert.Contains(t, logs.All()[0].Message, "cannot get singular result")
+
+	finalResult := &pipeline.FinalResult{
+		Values:      []interface{}{"123456"},
+		AllErrors:   nil,
+		FatalErrors: []error{nil},
+	}
+	obs = getObservation(ds, finalResult)
+	assert.Equal(t, obs, int64(123456))
+}
+
+func TestCollectAndSend(t *testing.T) {
+	ingressClient := mocks.NewTelemetryIngressClient(t)
+	ingressAgent := telemetry.NewIngressAgentWrapper(ingressClient)
+	monitoringEndpoint := ingressAgent.GenMonitoringEndpoint("0xa", synchronization.EnhancedEA)
+	ingressClient.On("Send", mock.AnythingOfType("synchronization.TelemPayload")).Return()
+
+	lggr, logs := logger.TestLoggerObserved(t, zap.WarnLevel)
+	ds := &inMemoryDataSource{
+		jb: job.Job{
+			ID:   1234567890,
+			Type: job.Type(pipeline.OffchainReportingJobType),
+			OCROracleSpec: &job.OCROracleSpec{
+				CaptureEATelemetry: true,
+			},
+		},
+		lggr:               lggr,
+		monitoringEndpoint: monitoringEndpoint,
+	}
+
+	finalResult := &pipeline.FinalResult{
+		Values:      []interface{}{"123456"},
+		AllErrors:   nil,
+		FatalErrors: []error{nil},
+	}
+
+	badTrrs := &pipeline.TaskRunResults{
+		pipeline.TaskRunResult{
+			Task: &pipeline.BridgeTask{
+				BaseTask: pipeline.NewBaseTask(0, "ds1", nil, nil, 0),
+			},
+			Result: pipeline.Result{
+				Value: nil,
+			},
+		}}
+
+	collectAndSend(ds, badTrrs, finalResult)
+	assert.Contains(t, logs.All()[0].Message, "cannot get bridge response from bridge task")
+
+	badTrrs = &pipeline.TaskRunResults{
+		pipeline.TaskRunResult{
+			Task: &pipeline.BridgeTask{
+				BaseTask: pipeline.NewBaseTask(0, "ds1", nil, nil, 0),
+			},
+			Result: pipeline.Result{
+				Value: "[]",
+			},
+		}}
+	collectAndSend(ds, badTrrs, finalResult)
+	assert.Equal(t, logs.Len(), 3)
+	assert.Contains(t, logs.All()[1].Message, "cannot parse EA telemetry")
+	assert.Contains(t, logs.All()[2].Message, "cannot get json parse value")
+}
+
+func BenchmarkCollectEATelemetry(b *testing.B) {
+	wg := sync.WaitGroup{}
+	ingressClient := mocks.NewTelemetryIngressClient(b)
+	ingressAgent := telemetry.NewIngressAgentWrapper(ingressClient)
+	monitoringEndpoint := ingressAgent.GenMonitoringEndpoint("0xa", synchronization.EnhancedEA)
+
+	ingressClient.On("Send", mock.AnythingOfType("synchronization.TelemPayload")).Return().Run(func(args mock.Arguments) {
+		wg.Done()
+	})
+
+	ds := inMemoryDataSource{
+		jb: job.Job{
+			Type: job.Type(pipeline.OffchainReportingJobType),
+			OCROracleSpec: &job.OCROracleSpec{
+				ContractAddress:    ethkey.EIP55AddressFromAddress(utils.RandomAddress()),
+				CaptureEATelemetry: true,
+				EVMChainID:         (*utils.Big)(big.NewInt(9)),
+			},
+		},
+		lggr:               nil,
+		monitoringEndpoint: monitoringEndpoint,
+	}
+	finalResult := pipeline.FinalResult{
+		Values:      []interface{}{"123456"},
+		AllErrors:   nil,
+		FatalErrors: []error{nil},
+	}
+	wg.Add(b.N)
```

Comment:
i believe N is determined dynamically while running, based on how many loops can be executed in a given time. this is what i saw while writing the json benchmarks, but a i can't find a definitive reference ATM.

would be better to do wg.Add(1) inside the loop

Reply:
You are right, after some research I found this in the documentation:

> During benchmark execution, b.N is adjusted until the benchmark function lasts long enough to be timed reliably.

Changed to `wg.Add(1)` in [`3214452` (#8684)](https://github.com/smartcontractkit/chainlink/pull/8684/commits/3214452890eb540bf560dabd5f210b65a68697af)